packages/ui/src/components/va-tree-view/hooks/useTreeBuilder.ts

```diff
@@ -0,0 +1,68 @@
+import { TreeNode, CreateNodeFunc, UseTreeBuilderFunc, TreeBuilderFunc } from '../types'
+import { computed, reactive, ref } from 'vue'
+
+export const createNode: CreateNodeFunc = ({
+  node,
+  children = [],
+  level = 0,
+  filter = ref(''),
+  labelKey = ref(''),
+  expandAll = ref(false),
+}) => (reactive({
+  ...node,
+  children,
+  level,
+  hasChildren: !!children.length,
+  expanded: expandAll.value || node.expanded || false,
+  checked: node.checked || false,
+  disabled: node.disabled || false,
+  matchesFilter: computed(() => {
+    if (!filter.value) { return true }
+
+    return node[labelKey.value].toLowerCase().includes(filter.value)
+  }).value,
+}))
+
+const useTreeBuilder: UseTreeBuilderFunc = (props) => {
+  const { nodes, expandAll, filter, labelKey } = props
+
+  const buildTree: TreeBuilderFunc = (nodes: TreeNode[], level = 0) => {
+    return nodes.reduce((acc: TreeNode[], node: TreeNode) => {
```

Comment:
I'm a bit confused about nodes when somewhere it ref somewhere not 😅. 

Please, remove argument for this FN. And it actually can be moved to computed I see.
```suggestion
    return nodes.value.reduce((acc: TreeNode[], node: TreeNode) => {
```